pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -438,6 +439,46 @@ public BrokerResponse handleRequest(JsonNode request, @Nullable RequesterIdentit
     return brokerResponse;
   }
 
+  /**
+   * Check if table is in the format of [database_name].[table_name].
+   *
+   * Only update TableName in QuerySource if there is no existing table in the format of [database_name].[table_name],
+   * but only [table_name].
+   *
+   * @param brokerRequest
+   */
+  private void updateQuerySource(BrokerRequest brokerRequest) {
+    String tableName = brokerRequest.getQuerySource().getTableName();
+    // Check if table is in the format of [database_name].[table_name]
+    String[] querySourceSplits = tableName.split("\\.", 2);
+    if (querySourceSplits.length != 2) {
+      return;
+    }
+    // Update table name if there is no existing table in the format of [database_name].[table_name] but only [table_name]
+    if (_enableCaseInsensitive && _tableCache.existTableName(querySourceSplits[1]) && !_tableCache
+        .existTableName(tableName)) {
+      // Use TableCache to check case insensitive table name.
+      brokerRequest.getQuerySource().setTableName(querySourceSplits[1]);
+      return;
+    }
```

Comment:
Use 2 if checks to prevent using RoutingManager for case insensitive table
```suggestion
    if (_enableCaseInsensitive) {
      // Use TableCache to check case insensitive table name.
      if (_tableCache.existTableName(querySourceSplits[1]) && !_tableCache.existTableName(tableName)) {
        brokerRequest.getQuerySource().setTableName(querySourceSplits[1]);
      }
      return;
    }
```